File: week_17/D1/python-vs-js/sample.js
// Ternary Operators
let pythonRules = true;
// pythonRules ? console.log("python really rules") : print("python still rules");

// if conditionals
// if (pythonRules) {
//   console.log("i'm already tired of typing");
// } else {
//   console.log("some words");
// }

// multiline comments

/* 
 this is a multiline
 comment
*/

// Random numbers

let teachers = ["Andrew", "David", "Krishna"];
let randomIdx = Math.floor(Math.random() * teachers.length);
// console.log(randomIdx);
// console.log(teachers[randomIdx]);

// get input from a user

// const readline = require("readline");
// const rl = readline.createInterface({
//   input: process.stdin,
//   output: process.stdout,
// });
//
// rl.question("What is your name, friend? ", (ans) => {
//   console.log(`Welcome to module 6 ${ans}. I hope you like snakes!`);
//   rl.close();
// });

// filter method and fat arrow syntax
let nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
let odds = nums.filter((x) => x % 2 !== 0);
// console.log(odds);

// classes
class Student {
  constructor(name, pythonKnowledge) {
    this.name = name;
    this.pythonKnowledge = pythonKnowledge;
  }
  getStudentInfo() {
    return `${this.name} knows ${this.pythonKnowledge} about python!`;
  }
}

const teagan = new Student((name = "Teagan"), (pythonKnowledge = "jack"));
console.log(teagan.getStudentInfo());

app.get("/", (req, res) => {
  res.send("<h1>Welcome to my express app!!!!!</h1>");
});

File: week_17/D1/python-vs-js/sample.py
# Ternary Operators
python_rules = False
# print("python is better") if python_rules else print("python is still better")

# if condtionals
# if python_rules:
#     print("python is better")
#     print("because it just is")
# else:
#     print("javscript honestly isn't that bad either")
#     print("but python is still better")

# multiline strings and comments

multi_line_str = """
this is a 
multiline string!!!!
"""

# print(multi_line_str)

"""
This is a multiline
comment
"""

# random numbers
import random
teachers = ["Andrew", "David", "Krishna"];
random_idx = random.randint(0, len(teachers) - 1)
# print(teachers[random_idx])
# print(dir(random))
# print(random.choice(teachers))


# asking for user input

# name = input("What is your name, traveler? ")
# print(f"Welcome to mod 6, {name}. It's going to be fun")

# filter built-in and lambda syntax
nums = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]

def get_odds(x):
    return x % 2 != 0

# odds = filter(lambda x: x % 2 != 0, nums)
odds = filter(get_odds, nums)

# lambda_get_odds = lambda x: x % 2 != 0
# odds = filter(lambda_get_odds, nums)

# print(list(odds))

# classes
class Student:
    def __init__(self, name, python_knowledge):
        self.name = name
        self.python_knowledge = python_knowledge

    def get_student_info(self):
        return f"{self.name} knows {self.python_knowledge} about python!"

andres = Student("Andres", "a good amount")

print(andres.get_student_info())


